### DISCUSSION TITLE
Deleting a resource from Template when using CDK Override?

### DISCUSSION THREAD

**Discussion by chadinwork** [aws/copilot-cli#5149] [Q&A] [answered]
Referencing the documentation here: https://aws.github.io/copilot-cli/docs/developing/overrides/cdk/

If we would like to delete a Resource from the template (`this.template`) how do we do that?

The documentation example shows us how to add new resources to a template, and how to delete a resource property, but not how to delete a resource.

Thank you!

**Comment by KollaAdithya** [comment]
Hello @chadinwork !

For example this sample adds a property to `EnvironmentMangerRole`
<details> <summary>CDK overrides sample</summary> 
import * as cdk from 'aws-cdk-lib';
import * as path from 'path';
import { aws_iam as iam } from 'aws-cdk-lib';

interface TransformedStackProps extends cdk.StackProps {
    readonly appName: string;
    readonly envName: string;
}

export class TransformedStack extends cdk.Stack {
    public readonly template: cdk.cloudformation_include.CfnInclude;
    public readonly appName: string;
    public readonly envName: string;

    constructor (scope: cdk.App, id: string, props: TransformedStackProps) {
        super(scope, id, props);
        this.template = new cdk.cloudformation_include.CfnInclude(this, 'Template', {
            templateFile: path.join('.build', 'in.yml'),
        });
        this.appName = props.appName;
        this.envName = props.envName;
        this.transformEnvironmentManagerRole();
    }

    transformEnvironmentManagerRole() {
        const environmentManagerRole = this.template.getResource("EnvironmentManagerRole") as iam.CfnRole;
        if (this.envName === "prod") {
            let assumeRolePolicy = environmentManagerRole.assumeRolePolicyDocument
            let statements = assumeRolePolicy.Statement
            statements.push({
                 "Effect": "Allow",
                 "Principal": { "Service": "ec2.amazonaws.com" },
                 "Action": "sts:AssumeRole"
            })
        }
    }
}

</details>

If you want to delete this property added to `EnvironmentManagerRole` you can just remove the `transformEnvironmentManagerRole()` on the CDK construct.

If you run `copilot env deploy --name prod` this will be delete the property set as a part of `CDK overrides` on `EnvironmentMangerRole`.

**Reply by chadinwork** [reply]
I'm not looking to delete a **property**. I'm looking to delete a **resource**. Such as delete a Cloudwatch Log Group.

**Comment by KollaAdithya** [comment]
Hi @chadinwork !
For example with this cdk L1 construct I am creating a new log group from CDK overrides in the service CloudFormation stack. 
```
import * as cdk from 'aws-cdk-lib';
import * as logs from 'aws-cdk-lib/aws-logs';
import * as path from 'path';

interface TransformedStackProps extends cdk.StackProps {
  readonly appName: string;
  readonly envName: string;
}

export class TransformedStack extends cdk.Stack {
  public readonly template: cdk.cloudformation_include.CfnInclude;
  public readonly appName: string;
  public readonly envName: string;

  constructor(scope: cdk.App, id: string, props: TransformedStackProps) {
    super(scope, id, props);
    this.template = new cdk.cloudformation_include.CfnInclude(this, 'Template', {
      templateFile: path.join('.build', 'in.yml'),
    });
    this.appName = props.appName;
    this.envName = props.envName;
    this.addnewLoggroupResource();
  }

  addnewLoggroupResource() {
    // Create a new LogGroup
    const newLogGroup = new logs.LogGroup(this, 'MyLogGroup', {
      logGroupName: this.appName+"LogGroup",
      retention: logs.RetentionDays.FIVE_MONTHS,
    });
  }
}
```
  After just removing the `this.addnewLoggroupResource()`  and running `copilot svc deploy` again will just delete the log group created.  let me know if this helps!

**Comment by KollaAdithya** [comment, marked as answer]
May be i misinterpreted the question. sorry for all the churn. 
If you want to delete the default `LogGroup` created by Copilot you can use this override
```
import * as cdk from 'aws-cdk-lib';
import * as path from 'path';

interface TransformedStackProps extends cdk.StackProps {
  readonly appName: string;
  readonly envName: string;
}

export class TransformedStack extends cdk.Stack {
  public readonly template: cdk.cloudformation_include.CfnInclude;
  public readonly appName: string;
  public readonly envName: string;

  constructor(scope: cdk.App, id: string, props: TransformedStackProps) {
    super(scope, id, props);
    this.template = new cdk.cloudformation_include.CfnInclude(this, 'Template', {
      templateFile: path.join('.build', 'in.yml'),
    });
    this.appName = props.appName;
    this.envName = props.envName;
    this.template.node.tryRemoveChild("LogGroup")
  }
}
```

**Comment by KollaAdithya** [comment]
I also created this PR (#5153) that will update our docs to showcase this example on deletion of a resource.

**Comment by chadinwork** [comment]
Thank you so much!
